test/compilable/test18295.d

```diff
@@ -0,0 +1,19 @@
+// REQUIRED_ARGS: -dip1000
+
+// https://issues.dlang.org/show_bug.cgi?id=18295
+
+// See test/fail_compilation/test18295.d for the non-`-dip1000` version`
+
+scope class C { int i; }    // Notice the use of `scope` here
+
+C increment(C c)            // Should not emit error here
+{
+    c.i++;
+    return c;
+}
+
+void main()
+{
+    scope C c = new C();
+    c.increment();
```

Comment:
Should error here, cannot pass scope to non-scope parameter `c`.
```
bug.d(10): Error: scope variable `c` assigned to non-scope parameter `c` calling bug.increment
```

Reply:
Added `return scope` to the parameter so there should be no error.

Reply:
Added `return scope` to the parameter, there should be no error.